### ISSUE TITLE
Implement EGLNativeSurface for RawWindowHandle

### ISSUE THREAD

**Issue by i509VCB** [Smithay/smithay#663]
Smithay at the moment can't integrate with any windowing library that implements RawWindowHandle.

**Comment by Drakulix** [comment]
Yeah those traits kinda pre-date `raw-window-handle`. Compatibility here would obviously be nice.
